### PR TITLE
Add support for custom render engines

### PR DESCRIPTION
**Pull request by JShep1**
See https://github.com/ignitionrobotics/ign-rendering/pull/142 for the PR adding this functionality.

Signed-off-by: John Shepherd <john@openrobotics.org>

### CODE REVIEW

**Review by iche033 — APPROVED** [review]

**Review by chapulina — CHANGES_REQUESTED** [review]
Works for me! I tested with `ign-rendering`'s hello world example like this:

```
cd src/ign-rendering/examples/hello_world_plugin/build
IGN_GAZEBO_RENDER_ENGINE_PATH=`pwd` ign gazebo -v 2 sensors_demo.sdf --render-engine HelloWorldPlugin
```



Mind adding a tutorial similar to https://github.com/ignitionrobotics/ign-gazebo/blob/main/tutorials/physics.md?

I think you can get most of the way there by copying that tutorial and `s/physics/rendering`

**Comment by chapulina** [comment]
Removed the "needs upstream release" label because `ign-rendering 3.2.0` was released. 

Mind updating the minimum required version here?

https://github.com/ignitionrobotics/ign-gazebo/blob/0d16dc96ec0544440a082f37d2cfa72a97dcbc9f/CMakeLists.txt#L117

File: src/rendering/RenderUtil.cc [hunk 1]
```diff
@@ -188,6 +188,9 @@ class ignition::gazebo::RenderUtilPrivate
   /// \brief Whether the transform gizmo is being dragged.
   public: bool transformActive{false};
 
+  /// \brief Environment variable which holds paths to look for engine plugins.
+  public: std::string pluginPathEnv = "IGN_GAZEBO_RENDER_ENGINE_PATH";
```
[the remark this answers is not in the document]
Comment (JShep1) [reply src/rendering/RenderUtil.cc]: 2e0e9f1eea6c9da542af501e3b95d55fa3bd1f37

**Comment by JShep1** [comment]
> Mind adding a tutorial similar to https://github.com/ignitionrobotics/ign-gazebo/blob/main/tutorials/physics.md?

Yep! Tutorial is [here](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering3/tutorials/03_rendering_plugins.md) and an example is [here](https://github.com/ignitionrobotics/ign-rendering/tree/ign-rendering3/examples/hello_world_plugin)

**Review by chapulina — APPROVED** [review]

**Pull request merged**